src/__tests__/App.test.jsx

```diff
@@ -1,8 +1,20 @@
 import App from '../App.jsx'
-import { describe, it, expect } from 'vitest';
+import { describe, it, beforeEach, afterEach, expect, vi } from 'vitest';
 import { render, fireEvent, screen } from "@testing-library/react";
+// Extra tests:
+// bday must be valid date - day can't be above 31/30/29 depending on month
+
 
 describe("App component", () => {
+    beforeEach(() => {
+        vi.useFakeTimers()
+        const todayDate = new Date('2024-01-01')
+        vi.setSystemTime(todayDate)
+      })
+    
+    afterEach(() => {
+        vi.useRealTimers()
+      })
   it("display Greeting", () => {
```

Comment:
Did you know you can put nested describe() statements. So you can have the root describe say "App component" like you already have but then you could wrap these next for tests in another describe with something like "Initial Render" or something like that which would nicely group them so people know they're just checking the initial rendering of the page. Then you can wrap the other tests in another describe with something like "Updating Birthday" or something like that!

Reply:
javascript also has parameterised tests!